macros/timeHistoryAnalyzer.py

```diff
@@ -0,0 +1,467 @@
+
+    Returns:
+        A list of MaskedRange objects
+    """
+    ranges = _findRangesMeta(data,
+                             vfat,
+                             channel,
+                             data.mask[vfat,channel] != 0,
+                             maxSkip)
+
+    return list(filter(lambda r: r.maskedScanCount() >= minBadScans,
+                       ranges))
+
+def findRangesNoise(data,
+                    vfat,
+                    channel,
+                    maxNoise = 0.02,
+                    maxSkip = 5,
+                    minBadScans = 4):
+    """Finds ranges of scans based on noise.
+
+    Searches the data for the given vfat and channel for ranges of low-noise
+    scans. A scan has low noise if noise < maxNoise. During the search, at most
+    maxSkip scans with mask not set can be skipped. Only ranges with more than
+    minBadScans are kept.
+
+    Args:
+        data: The TimeSeriesData object to pull data from
+        vfat: The VFAT to return ranges for
+        channel: The channel to return ranges for
+        maxNoise: Scans with a noise below this value are considered low-noise
+        maxSkip: The maximum number of consecutive scans that don't fulfill the
+            low-noise condition between two scans that do
+        minBadScans: The minimum number of low-noise scans in any returned range
+
+    Returns:
+        A list of MaskedRange objects
+    """
+    ranges = _findRangesMeta(data,
+                             vfat,
+                             channel,
+                             data.noise[vfat,channel] < maxNoise,
+                             maxSkip)
+
+    return list(filter(lambda r: np.count_nonzero(r.noise() < maxNoise) >= minBadScans,
+                       ranges))
+
+class TimeSeriesData(object):
+    """Holds information about time variation of scan results.
+
+    Each property is stored as a 3D Numpy array with indexes
+    [vfat][channel][time]. The time index -> scan date mapping is exposed in the
+    date attribute.
+
+    Attributes:
+        dates: Numpy array of strings containing the scan dates.
+        mask: Data for the "mask" property,
+        maskReason: Data for the "maskReason" property,
+    """
+
+    def __init__(self, inputDir):
+        """Creates a TimeSeriesData object by reading the files located in the
+        inputDir directory.
+
+        The input directory must contain the following files:
+
+        * gemPlotterOutput_mask_vs_scandate.root
+        * gemPlotterOutput_maskReason_vs_scandate.root
+        * gemPlotterOutput_noise_vs_scandate.root
+
+        They are created by plotTimeSeries.py.
+
+        Args:
+            inputDir: The path to the input directory
+        """
+        import ROOT as r
+        from root_numpy import hist2array
+
+        file_mask = r.TFile('%s/gemPlotterOutput_mask_vs_scandate.root' % inputDir, 'READ')
+        if file_mask.IsZombie():
+            raise IOError('Could not open %s' % file_mask.GetPath())
```

Comment:
Explicitly tell the user with this error message they were supposed give the `$PWD` which is the output of `plotTimeSeries.py`.